### ISSUE TITLE
AMQPConnection->disconnect() not closing connection on rabbitmq server

### ISSUE THREAD

**Issue by J-Siu** [pdezwart/php-amqp#183]
Not sure if this is a rabbitmq server side issue or php-amp issue.

I have following code to compare new connection vs re-using connection. However I discovered that the connection on rabbitmq is not closed, though disconnect() return successful.

```
<?php

include (__DIR__ .'/JS-amqp-include.php');

$reuse_connection = false;
$cycle            = 5000;
$input            = "test";
$mypid            = getmypid();

//	$msg_body = implode(' ', array_slice($argv, 1));
echo "PID: $mypid\n";

if ($reuse_connection) {
	$setup_start = microtime(true);

	$co = new AMQPConnection(array('host' => HOST, 'port' => PORT, 'vhost' => VHOST, 'login' => USER, 'password' => PASS));
	$co->connect();
	$ch = new AMQPChannel($co);
	$ex = new AMQPExchange($ch);
	$ex->setName(X_DIR);

	$setup_end      = microtime(true);
	$setup_duration = $setup_end-$setup_start;
	echo "Setup Duration: $setup_duration Sec.\n";
}

$time_start = microtime(true);

for ($i = 1; $i <= $cycle; $i++) {
	if (!$reuse_connection) {
		$co = new AMQPConnection(array('host' => HOST, 'port' => PORT, 'vhost' => VHOST, 'login' => USER, 'password' => PASS));
		$co->connect();
		$ch = new AMQPChannel($co);
		$ex = new AMQPExchange($ch);
		$ex->setName(X_DIR);
	}
	$ex->publish("$mypid:$input:$i");

	if (!$reuse_connection) {
		if ($co->disconnect()) {echo "Disconnected\n";} else {echo "Disconnect Failed.\n";}
	}

}

$time_end      = microtime(true);
$time_duration = $time_end-$time_start;
echo "Duration: $time_duration Sec.\n";

?>
```

Following is netstat out put from rabbitmq server after the above program is stopped.
```
tcp6       0      0 192.168.0.111:5672      192.168.0.114:33350     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:32772     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:60634     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:60836     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:32974     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:33574     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:32978     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:60406     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:32852     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:33582     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:60672     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:32872     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:60834     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:33176     ESTABLISHED
tcp6       0      0 192.168.0.111:5672      192.168.0.114:60658     ESTABLISHED
$ netstat -na|grep  ^tcp|wc -l
841
```

**Comment by pinepain** [comment]
Looks like this is already fixed in 1.7.0 (check 1.7.0alpha1). I would recommend you to check and install the latest version (even unstable).

I can reproduce this with php-amqp 1.6.0, but not with 1.7.0. Probably, this bug was fixed within PR https://github.com/pdezwart/php-amqp/pull/181.

For further discussion, please, specify your OS, PHP, php-amqp, RabbitMQ versions and a way they were installed (see [How to report a problem](https://github.com/pdezwart/php-amqp#how-to-report-a-problem) section in readme).

**Comment by J-Siu** [comment]
I installed 1.7.0 and the issue is fixed. Closing now. Thank you!!!
